app/src/features/Preview/Preview.tsx

```diff
@@ -141,28 +161,40 @@ const Preview = (): JSX.Element => {
               <TableRow hover key={uuid()} className={clsx(classes.rowBorder)}>
                 <TableCell
                   className={clsx(classes.fhirIconCell, classes.cellsTitle)}
-                  onClick={() => handleFhirIconClick(index)}
+                  onClick={handleFhirIconClick(index)}
                 >
                   <FhirIconCell />
                 </TableCell>
-                {columnData.map((an, i) => (
+                {columnData.map((cell, i) => (
                   <TableCell className={classes.cells} key={i}>
-                    {an}
+                    {cell}
                   </TableCell>
                 ))}
               </TableRow>
             ))}
           </TableBody>
-          <TableFooter></TableFooter>
         </Table>
       </TableContainer>
-      <div className={classes.texts}>
-        <Typography>
-          {t("clickOnAFireIcon")}{" "}
-          <Icon icon={IconNames.FLAME} className={classes.iconFlame} />{" "}
-          {t("inOrderToPreview")}
-        </Typography>
-      </div>
+      {!preview && (
+        <div className={classes.texts}>
+          <Typography>
+            {t("clickOnAFireIcon")}{" "}
+            <Icon icon={IconNames.FLAME} className={classes.iconFlame} />{" "}
+            {t("inOrderToPreview")}
+          </Typography>
+        </div>
+      )}
+      {preview && (
+        <div className={classes.preview}>
+          <ReactJson
+            src={preview}
+            theme={prefersDarkMode ? "summerfruit" : "summerfruit:inverted"}
+            collapsed={1}
+            displayObjectSize={false}
+            displayDataTypes={false}
+          />
+        </div>
+      )}
```

Comment:
maybe a ternary would be clearer ? eg:
`{preview ? <ReactJson> : <Typography>}`